### PR TITLE
BinarySensor `ignore_internal_state` only on GroupValueWrite

### PR DESCRIPTION
**Pull request by farmio**
BinarySensor:
- `ignore_internal_state` defaults to False again (like HA config default)
- `ignore_internal_state` and counter are only applied to GroupValueWrite telegrams, not GroupValueRespond.
- if `context_timeout` is set `ignore_internal_state` is set to True.

### CODE REVIEW

**Review by marvin-w — CHANGES_REQUESTED** [review]
Looks fine, please add the missing tests.

File: test/devices_tests/sensor_test.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ async def async_after_update_callback(device):
         sensor.sensor_value.payload = payload
 
         telegram = Telegram(group_address=GroupAddress("1/2/3"), payload=payload)
+        response_telegram = Telegram(
```
Comment (marvin-w) [inline test/devices_tests/sensor_test.py:82]: Why did you change this test?

Comment (farmio) [reply test/devices_tests/sensor_test.py:82]: So it can use process() instead of directly process_group_response() like it is done in production by the TelegramQueue. (line 101)

**Review by marvin-w — APPROVED** [review]
Lovely!
